### DISCUSSION TITLE
video21

### DISCUSSION THREAD

**Discussion by mcharlie9** [jacekschae/learn-pedestal-course-files#13] [Feedback]
In addition to the discussion under video14, please find below further feedbacks regarding video21: 

1) Having gone through the interceptor section (video16-18) and just learnt how to add intercepters using the service-map, one would expect the interceptor-db to be added in a similar fashion. However, interceptor-db is to be added directly to the routes. For students new to the concept of interceptors as well as pedestal's way of making using of them, a little explanation as to how interceptor-db is different from the other interceptors; why it is added directly to the routes; what the benefits are, etc. might provide a better transition.

2) The pedestal test is very useful. However, while it shows that the routes are working, it does not show clearly that the interceptor-db is being added properly.  Maybe a way to ascertain the proper addition of interceptor-db would wrap this video up nicely.

3) This is just a side comment. Development environment serves on localhost, and since the :host key is optional and defaults to localhost anyway, maybe a simpler way is to leave it out of the (def routes ...).  Same case for :scheme as it is also optional.  Of course, deployment would be another story, and you can circle back to how :host works at that time if it turns out to matter.

Other than the above, all is very good.  I had fun going through the new sets of videos.  These videos are not easy to make, so kudos to you and well done. Thanks again for making this challenging course!

**Comment by jacekschae** [comment]
Thanks a lot. I think I'll do as you recommended for point 3 -- I was thinking the same. 

When it comes to points nr. 2 It's hard to capture everything all the aspects in the videos therefore video 22 will be something where we'll look into that. 

Last but not least. Good point. I'll try to explain why adding db at the same time as conn would be a bad idea -- if we would do that at the same time as connection our db value would be always the value when the system starts -- all the changes would happen but if the value of db would be the one that was there at the start.

**Comment by jacekschae** [comment]
Just uploaded 21-response-for (take 2)

**Comment by mcharlie9** [comment]
This take 2 is excellent, the explanations are spot-on, well done!  👍
